### PR TITLE
Loader fix 2(my_expense, my_reports)

### PR DESCRIPTION
**Pull request by iamtarun1993**

### CODE REVIEW

File: src/app/fyle/my-expenses/my-expenses.page.ts [hunk 1]
```diff
@@ -208,6 +212,10 @@ export class MyExpensesPage implements OnInit {
       pageNumber: 1
     });
 
+    this.params$ = this.loadData$.asObservable().pipe(
+      skip(1)
```
Comment (Julias0) [inline src/app/fyle/my-expenses/my-expenses.page.ts:216]: why are we skipping the first?
If this is due to initialization of BehaviorSubject, we can just replace BehaviorSubject with Subject

File: src/app/fyle/my-expenses/my-expenses.page.ts [hunk 2]
```diff
@@ -204,9 +196,31 @@ export class MyExpensesPage implements OnInit {
     this.simpleSearchText = '';
 
     this.currentPageNumber = 1;
-    this.loadData$ = new BehaviorSubject({
-      pageNumber: 1
-    });
+
+    if (this.activatedRoute.snapshot.queryParams.filters) {
```
Comment (Julias0) [inline src/app/fyle/my-expenses/my-expenses.page.ts:200]: lets extract this whole if else block into a function

File: src/app/fyle/my-reports/my-reports.page.ts [hunk 3]
```diff
@@ -96,9 +90,27 @@ export class MyReportsPage implements OnInit {
     this.acc = [];
 
     this.currentPageNumber = 1;
-    this.loadData$ = new BehaviorSubject({
-      pageNumber: 1
-    });
+    if (this.activatedRoute.snapshot.queryParams.filters) {
```
Comment (Julias0) [inline src/app/fyle/my-reports/my-reports.page.ts:93]: this is the same in my expenses also. Can we think of a way to reduce duplication

Comment (iamtarun1993) [reply src/app/fyle/my-reports/my-reports.page.ts:93]: 2nd `else if` is different from `my expenses`. Will think of a better way later.
will put a //todo for now. Too much delay happening here

File: src/app/fyle/my-expenses/my-expenses.page.ts [hunk 4]
```diff
@@ -21,6 +21,7 @@ import {StorageService} from '../../core/services/storage.service';
 import { TokenService } from 'src/app/core/services/token.service';
 import { ApiV2Service } from 'src/app/core/services/api-v2.service';
 import { environment } from 'src/environments/environment';
+type myExpensesParams = {pageNumber: number, queryParams: any, sortParam: string, sortDir: string, searchString: string};
```
Comment (Julias0) [inline src/app/fyle/my-expenses/my-expenses.page.ts:24]: make this multiline

**Review by Julias0 — APPROVED** [review]

**Pull request merged**